### PR TITLE
Spike positions

### PR DESCRIPTION
**Pull request by maxnoe**
This adds functions to obtain the positions of the spikes needed for calculating the spike heights to enable spike subtraction instead of interpolation and a function to update the last_readout_times needed to create the time calibration file in this lstchain PR here: https://github.com/cta-observatory/cta-lstchain/pull/725

### CODE REVIEW

File: ctapipe_io_lst/calibration.py [hunk 1]
```diff
@@ -556,73 +556,108 @@ def interpolate_spike_A(waveform, position):
...
 
+    positions = []
+
     for k in range(4):
         # looking for spike A first case
         abspos = N_CAPACITORS_CHANNEL + 1 - N_SAMPLES - 2 - last_fc + k * N_CAPACITORS_CHANNEL + N_CAPACITORS_PIXEL
         spike_A_position = (abspos - current_fc + N_CAPACITORS_PIXEL) % N_CAPACITORS_PIXEL
 
-        if 2 < spike_A_position < (N_SAMPLES - 2):
+        # a spike affects the poistion itself and the two following slices
```
Comment (FrancaCassol) [inline ctapipe_io_lst/calibration.py:584]: typo: "position"

**Review by FrancaCassol — APPROVED** [review]

File: ctapipe_io_lst/calibration.py (hunk 1 above)
Comment (maxnoe) [reply ctapipe_io_lst/calibration.py]: fixed

**Comment by maxnoe** [comment]
@jsitarek sorry, I don't see any comment, can you post it here again or open an issue?

**Comment by jsitarek** [comment]
I started the review and see it above, maybe you do not see it because I did not finalize the review. Either way this is my comment (to line 586):
"

this looks suspicious to me (there are 2 more similar changes below). First of all spike_A_position comes from modulo division, so it cannot be -1, hence the first condition will always be true.
Also, while for spike subtraction it should be fine, if the interpolation is done it has to be modified as well. You cannot interpolate the first slice (you can only extrapolate it by taking the value of the 2nd slice), also if the interpolation is done before subtraction of the noisy slices this will cause further problems. If I remember correcty first 3 slices and the last one slice is noisy, so they should not be used for interpolation.

@pawel21 can you also have a look at this?
"

**Comment by maxnoe** [comment]
> I started the review and see it above, maybe you do not see it because I did not finalize the review. Either way this is my comment (to line 586):

Yes you need to submit it. Others cannot see "pending" comments.

**Comment by maxnoe** [comment]
> You cannot interpolate the first slice (you can only extrapolate it by taking the value of the 2nd slice)

The calculation of the positions is independent of the application. The interpolation code checks that the calculated position is > 2, as before.

**Comment by maxnoe** [comment]
But you are right about the negative issue with modulus... I assumed C like behaviour (where -2 % 1024 == -2)

**Pull request merged**